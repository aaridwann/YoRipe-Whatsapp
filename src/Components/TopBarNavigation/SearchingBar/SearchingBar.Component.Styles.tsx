import { StyleSheet } from 'react-native';

import Constants from '../../../Constants/Constants';
import { scaleSize, scaleWidth } from '../../../Utils/Size/size.utils';

export const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        width: '100%',
        padding: 5,
        backgroundColor: Constants.Colors.GREEN_MEDIUM_DARK,
        borderRadius: 20,
    },
    inputContainer: {
        minWidth: scaleWidth(260),
        padding: scaleSize(6),
        borderRadius: 8,
    },
    input: {
        color: Constants.Colors.WHITE,
    },
});
