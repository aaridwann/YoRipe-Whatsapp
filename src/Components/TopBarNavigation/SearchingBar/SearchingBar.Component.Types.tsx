export interface Props {
    isShow: boolean;
    onSearch: (val: string) => void;
    onBack: VoidFunction;
}
