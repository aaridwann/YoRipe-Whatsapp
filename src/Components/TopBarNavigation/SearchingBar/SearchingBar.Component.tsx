import React from 'react';
import { TextInput, TouchableOpacity, View } from 'react-native';
import * as Animatable from 'react-native-animatable';
import Ionicons from '@expo/vector-icons/Ionicons';

import Constants from '../../../Constants/Constants';
import { scaleSize } from '../../../Utils/Size/size.utils';
import { Props } from './SearchingBar.Component.Types';
import { styles } from './SearchingBar.Component.Styles';

const _renderInput = (props: Props) => (
    <View style={styles.inputContainer}>
        <TextInput
            accessibilityLabel="Navbar_Searching_Input"
            placeholder="Find a message"
            placeholderTextColor={'white'}
            onChangeText={(e) => props.onSearch(e)}
            style={styles.input}
        />
    </View>
);

const _renderBackIcon = (props: Props) => (
    <TouchableOpacity
        accessibilityLabel="Navbar_Searching_Back_Button"
        onPress={props.onBack}
        style={{ marginHorizontal: scaleSize(8) }}
    >
        <Ionicons
            name="arrow-back-outline"
            size={scaleSize(25)}
            color={Constants.Colors.WHITE}
        />
    </TouchableOpacity>
);

const _renderRightSide = () => (
    <View
        style={{
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center',
        }}
    >
        <TouchableOpacity style={{ marginHorizontal: scaleSize(8) }}>
            <Ionicons
                name="chevron-up"
                size={scaleSize(25)}
                color={Constants.Colors.WHITE}
            />
        </TouchableOpacity>
        <TouchableOpacity style={{ marginHorizontal: scaleSize(8) }}>
            <Ionicons
                name="chevron-down"
                size={scaleSize(25)}
                color={Constants.Colors.WHITE}
            />
        </TouchableOpacity>
    </View>
);

const _renderLeftSide = (props: Props) => (
    <View
        style={{
            flexDirection: 'row',
            justifyContent: 'center',
            alignItems: 'center',
        }}
    >
        {_renderBackIcon(props)}
        {_renderInput(props)}
    </View>
);

const SearchingBar = (props: Props) => {
    return (
        <Animatable.View
            accessibilityLabel="Navbar_Searching"
            animation={'bounceInDown'}
            style={styles.container}
        >
            {_renderLeftSide(props)}
            {_renderRightSide()}
        </Animatable.View>
    );
};

export default SearchingBar;
