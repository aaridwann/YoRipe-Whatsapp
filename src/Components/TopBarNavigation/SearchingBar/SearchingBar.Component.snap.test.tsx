import React from 'react';
import { render } from '@testing-library/react-native';

import SearchingBar from './SearchingBar.Component';

describe('Searching navbar ', () => {
    it('Should render with correct props', () => {
        const MockProps = {
            isShow: false,
            onSearch: jest.fn(),
            onBack: jest.fn(),
        };

        expect(render(<SearchingBar {...MockProps} />)).toMatchSnapshot();
    });
});
