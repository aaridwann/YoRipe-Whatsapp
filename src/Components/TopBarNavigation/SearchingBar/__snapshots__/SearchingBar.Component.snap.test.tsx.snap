// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searching navbar  Should render with correct props 1`] = `
<View
  accessibilityLabel="Navbar_Searching"
  collapsable={false}
  style={
    {
      "backgroundColor": "#054c44",
      "borderRadius": 20,
      "flexDirection": "row",
      "opacity": 0,
      "padding": 5,
      "transform": [
        {
          "translateY": -800,
        },
      ],
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityLabel="Navbar_Searching_Back_Button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginHorizontal": 16,
          "opacity": 1,
        }
      }
    >
      IonIcons
    </View>
    <View
      style={
        {
          "borderRadius": 8,
          "minWidth": 520,
          "padding": 12,
        }
      }
    >
      <TextInput
        accessibilityLabel="Navbar_Searching_Input"
        onChangeText={[Function]}
        placeholder="Find a message"
        placeholderTextColor="white"
        style={
          {
            "color": "#ffffff",
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginHorizontal": 16,
          "opacity": 1,
        }
      }
    >
      IonIcons
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginHorizontal": 16,
          "opacity": 1,
        }
      }
    >
      IonIcons
    </View>
  </View>
</View>
`;
